testcases/misc_tests/p11sak_test.sh

```diff
@@ -1338,6 +1341,469 @@ else
+	echo "* TESTCASE remove-cert x509 PASS Deleted imported base64-encoded x509 cert with EC-p384 public key"
+else
+	echo "* TESTCASE remove-cert x509 FAIL Failed to delete imported base64-encoded x509 cert with EC-p384 public key"
+	status=1
+fi
+
+# EC-p521
+grep -q "p11sak-x509-ecp521crt" $P11SAK_X509_PRE
+rc=$?
+if [ $rc = 0 ]; then
+	echo "* TESTCASE import-cert x509 PASS Imported binary x509 cert with random EC-p521 key"
+else
+	echo "* TESTCASE import-cert x509 FAIL Failed to import binary x509 certs with random EC-p521 key"
+	status=1
+fi
+grep -v -q "p11sak-x509-ecp521crt" $P11SAK_X509_POST
+rc=$?
+if [ $rc = 0 ]; then
+	echo "* TESTCASE remove-cert x509 PASS Deleted imported binary x509 cert with EC-p521 public key"
+else
+	echo "* TESTCASE remove-cert x509 FAIL Failed to delete imported binary x509 cert with EC-p521 public key"
+	status=1
+fi
+
+grep -q "p11sak-x509-ecp521pem" $P11SAK_X509_PRE
+rc=$?
+if [ $rc = 0 ]; then
+	echo "* TESTCASE import-cert x509 PASS Imported base64-encoded x509 cert with random EC-p521 key"
+else
+	echo "* TESTCASE import-cert x509 FAIL Failed to import base64-encoded x509 certs with random EC-p521 key"
+	status=1
+fi
+grep -v -q "p11sak-x509-ecp521pem" $P11SAK_X509_POST
+rc=$?
+if [ $rc = 0 ]; then
+	echo "* TESTCASE remove-cert x509 PASS Deleted imported base64-encoded x509 cert with EC-p521 public key"
+else
+	echo "* TESTCASE remove-cert x509 FAIL Failed to delete imported base64-encoded x509 cert with EC-p521 public key"
+	status=1
+fi
+
+# DSA-3072
+if [[ -n $( pkcsconf -m -c $SLOT | grep CKM_DSA) ]]; then
+	grep -q "p11sak-x509-dsa3072crt" $P11SAK_X509_PRE
+	rc=$?
+	if [ $rc = 0 ]; then
+		echo "* TESTCASE import-cert x509 PASS Imported binary x509 cert with random DSA-3072 key"
+	else
+		echo "* TESTCASE import-cert x509 FAIL Failed to import binary x509 certs with random DSA-3072 key"
+		status=1
+	fi
+	grep -v -q "p11sak-x509-dsa3072crt" $P11SAK_X509_POST
+	rc=$?
+	if [ $rc = 0 ]; then
+		echo "* TESTCASE remove-cert x509 PASS Deleted imported binary x509 cert with DSA-3072 public key"
+	else
+		echo "* TESTCASE remove-cert x509 FAIL Failed to delete imported binary x509 cert with DSA-3072 public key"
+		status=1
+	fi
+	
+	grep -q "p11sak-x509-dsa3072pem" $P11SAK_X509_PRE
+	rc=$?
+	if [ $rc = 0 ]; then
+		echo "* TESTCASE import-cert x509 PASS Imported base64-encoded x509 cert with random DSA-3072 key"
+	else
+		echo "* TESTCASE import-cert x509 FAIL Failed to import base64-encoded x509 certs with random DSA-3072 key"
+		status=1
+	fi
+	grep -v -q "p11sak-x509-dsa3072pem" $P11SAK_X509_POST
+	rc=$?
+	if [ $rc = 0 ]; then
+		echo "* TESTCASE remove-cert x509 PASS Deleted imported base64-encoded x509 cert with DSA-3072 public key"
+	else
+		echo "* TESTCASE remove-cert x509 FAIL Failed to delete imported base64-encoded x509 cert with DSA-3072 public key"
+		status=1
+	fi
+else
+	echo "Skip checking import-cert/remove-cert for certs with DSA public key, slot does not support CKM_DSA"
```

Comment:
For all the skip cases of the output and completion checking you MUST emit exactly the same number of messages as you emit for PASS or FAIL in the form
`	echo "* TESTCASE xxxxx SKIP description"`
so that it matches the corresponding PASS or FAIL messages. 
This is a requirement by the CI parser that parses this output. 

You may see the places for key test above.

Reply:
Done.